dace/sdfg/state.py

```diff
@@ -502,13 +503,22 @@ def _read_and_write_sets(self) -> Tuple[Dict[AnyStr, List[Subset]], Dict[AnyStr,
             # is read is not counted in the read set
             for n in utils.dfs_topological_sort(sg, sources=sg.source_nodes()):
                 if isinstance(n, nd.AccessNode):
-                    for e in sg.in_edges(n):
+                    in_edges = sg.in_edges(n)
+                    out_edges = sg.out_edges(n)
+                    # Filter out memlets which go out but the same data is written to the AccessNode by another memlet
+                    for out_edge in out_edges:
+                        for in_edge in in_edges:
+                            if in_edge.data.data == out_edge.data.data and \
```

Comment:
I suggest using parentheses rather than invertest slash, for example:
```python
if (in_edge.data.data == out_edge.data.data and
        in_edge.data.dst_subset.covers(out_edge.data.src_subset)):
    out_edges.remove(out_edge)
```